File: test-project/greeter.provider/src/main/java/examples/greeter/Friendly.java
package examples.greeter;

import examples.greeter.api.Greeter;
import java.io.*;
import java.util.*;

public class Friendly implements Greeter {
    @Override
    public String hello() {
        var stream = this.getClass().getResourceAsStream("/greeting.txt");
        try (BufferedReader reader = new BufferedReader(new InputStreamReader(stream, "utf-8"))) {
            return reader.readLine();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: test-project-kotlin-pre-1-7/greeter.provider/src/main/kotlin/examples/greeter/Friendly.kt
package examples.greeter

import examples.greeter.api.Greeter
import java.io.*

class Friendly: Greeter {
    override fun hello(): String {
        val stream = javaClass.getResourceAsStream("/greeting.txt")
        val reader = BufferedReader(InputStreamReader(stream, "utf-8"))
        return reader.readLine()
    }
}
